libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -1047,6 +1049,33 @@ public XContentBuilder copyCurrentStructure(XContentParser parser) throws IOExce
         return this;
     }
 
+    /**
+     * Used in {@link #writableFieldAsBase64(String, WritableValue)}
+     */
+    @FunctionalInterface
+    public interface WritableValue {
```

Comment:
Do we have Checked consumer here? I'm not sure the new interface adds any clarity. But if you think it does I'm certaily ok with it.

Reply:
We have it here, but I prefer not to expose the output stream to the caller.

Reply:
I changed my mind. I pushed https://github.com/elastic/elasticsearch/pull/73804/commits/1affa1973ddb2743cab85b7fedaa5f0ed126135c to remove the new interface and replace it with a CheckedConsumer.